flink-core/src/test/java/org/apache/flink/api/common/typeutils/SerializerTestBase.java

```diff
@@ -124,30 +124,8 @@ public void testSnapshotConfigurationAndReconfigure() throws Exception {
 		final TypeSerializer<T> serializer = getSerializer();
 		final TypeSerializerSnapshot<T> configSnapshot = serializer.snapshotConfiguration();
 
-		byte[] serializedConfig;
-		try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-			TypeSerializerSnapshotSerializationUtil.writeSerializerSnapshot(
-				new DataOutputViewStreamWrapper(out), configSnapshot, serializer);
-			serializedConfig = out.toByteArray();
-		}
-
-		TypeSerializerSnapshot<T> restoredConfig;
-		try (ByteArrayInputStream in = new ByteArrayInputStream(serializedConfig)) {
-			restoredConfig = TypeSerializerSnapshotSerializationUtil.readSerializerSnapshot(
-				new DataInputViewStreamWrapper(in), Thread.currentThread().getContextClassLoader(), getSerializer());
-		}
-
-		TypeSerializerSchemaCompatibility<T> strategy = restoredConfig.resolveSchemaCompatibility(getSerializer());
-		final TypeSerializer<T> restoreSerializer;
-		if (strategy.isCompatibleAsIs()) {
-			restoreSerializer = restoredConfig.restoreSerializer();
-		}
-		else if (strategy.isCompatibleWithReconfiguredSerializer()) {
-			restoreSerializer = strategy.getReconfiguredSerializer();
-		}
-		else {
-			throw new AssertionError("Unable to restore serializer with " + strategy);
-		}
+		final TypeSerializer<T> restoreSerializer = snapshotAndReconfigure(
```

Comment:
Is this a bug of this class? Is there any way do not touch this file?

Reply:
No. I just move these into a separate method so that it can be reused for my newly added tests.